src/transforms/transform-deployment.ts

```diff
@@ -170,12 +170,14 @@ const mapJiraIssueIdsAndCommitsToAssociationArray = (
 					repositoryId: repositoryId
 				};
 			});
-		associations.push(
-			{
-				associationType: "commit",
-				values: commitKeys
-			}
-		);
+		if (commitKeys.length) {
+			associations.push(
+				{
+					associationType: "commit",
+					values: commitKeys
+				}
+			);
+		}
```

Comment:
Hm, this code seems hard to read and could be more concise.  Maybe something like this instead?

```
const mapJiraIssueIdsAndCommitsToAssociationArray = (
    issueIds: string[],
    repositoryId: string,
    commitSummaries: CommitSummary[] = []
): JiraAssociation[] | undefined => {
    if (!issueIds?.length) {
        return undefined;
    }
    
    return [
        {
            associationType: "issueIdOrKeys",
            values: issueIds
        },
        {
            associationType: "commit",
            values: commitSummaries
                .slice(0, MAX_ASSOCIATIONS_PER_ENTITY - issueIds.length)
                .map((commitSummary) => ({
                    commitHash: commitSummary.sha,
                    repositoryId: repositoryId
                }));
        }
    ].filter(association => !!association.values?.length)
}
```

Reply:
But the commit association type should only exist if there are more than 0 commit keys after the slice has occurred.

Reply:
I actually found that harder to understand - but then again I am no expert in javascript.

Reply:
I have since tried the above - but the syntax checker in IDEA doesn't like something about the association type strings.

Reply:
What does it say?

Reply:
actually, nvm, we can fix that later :)